06-express-and-layered-architecture/src/cart/middleware/cart.middleware.ts

```diff
@@ -0,0 +1,24 @@
+import { getCart } from "../cart.repository";
+
+export async function calculateTotal(req: any, res: any, next: any) {
+  const userId = req.header("x-user-id");
+
+  if (req.path.endsWith("/checkout")) {
+    return next();
+  }
+
+  try {
+    const cart = await getCart(userId);
+    let total = 0;
+
+    console.log(cart.isDeleted);
+    if (cart && cart.items && !cart.isDeleted) {
+      cart.items.forEach((item) => (total += item.count * item.product.price));
+    }
+
+    res.locals.total = total;
```

Comment:
res.locals, while it works, serves a different purpose. it is used in case we return html generated by a view engine, such as pug or handlerbars using the `res.render()`. A view engine is like a templating engine (a bit like jsx in React) that can spit out html based on dynamic parameters, which dynamic parameters are made available as res.locals. In this use case, you api will return json to the client, so I would advise -if you want to keep this logic in a middleware to simply add a property to the request object: `req.total = total`.

For more information on locals, please check: https://expressjs.com/en/api.html#res.locals and https://stackoverflow.com/questions/24072333/difference-between-assigning-to-res-and-res-locals-in-node-js-express